File: legacy-runtime/src/main/java/com/xiaomi/channel/commonutils/misc/SerializedAsyncTaskProcessor.java
package com.xiaomi.channel.commonutils.misc;

import android.os.Handler;
import android.os.Looper;
import android.os.Message;
import com.xiaomi.channel.commonutils.logger.MyLog;
import java.util.concurrent.LinkedBlockingQueue;
import java.util.concurrent.TimeUnit;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/channel/commonutils/misc/SerializedAsyncTaskProcessor.class */
public class SerializedAsyncTaskProcessor {
    private static final int MSG_AFTER_EXECUTE = 1;
    private static final int MSG_BEFORE_EXECUTE = 0;
    private volatile SerializedAsyncTask mCurrentTask;
    private final boolean mIsDaemon;
    private int mKeepAliveTime;
    private Handler mMainThreadHandler;
    private ProcessPackageThread mProcessThread;
    private volatile boolean threadQuit;

    /* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/channel/commonutils/misc/SerializedAsyncTaskProcessor$ProcessPackageThread.class */
    private class ProcessPackageThread extends Thread {
        private static final String THREAD_NAME = "PackageProcessor";
        private final LinkedBlockingQueue<SerializedAsyncTask> mTasks;

        public ProcessPackageThread() {
            super(THREAD_NAME);
            this.mTasks = new LinkedBlockingQueue<>();
        }

        private void notifyUI(int i, SerializedAsyncTask serializedAsyncTask) {
            try {
                SerializedAsyncTaskProcessor.this.mMainThreadHandler.sendMessage(SerializedAsyncTaskProcessor.this.mMainThreadHandler.obtainMessage(i, serializedAsyncTask));
            } catch (Exception e) {
                MyLog.e(e);
            }
        }

        public void insertTask(SerializedAsyncTask serializedAsyncTask) {
            try {
                this.mTasks.add(serializedAsyncTask);
            } catch (Exception e) {
                e.printStackTrace();
            }
        }

        @Override // java.lang.Thread, java.lang.Runnable
        public void run() {
            long j = SerializedAsyncTaskProcessor.this.mKeepAliveTime > 0 ? SerializedAsyncTaskProcessor.this.mKeepAliveTime : Long.MAX_VALUE;
            while (!SerializedAsyncTaskProcessor.this.threadQuit) {
                try {
                    SerializedAsyncTask serializedAsyncTaskPoll = this.mTasks.poll(j, TimeUnit.SECONDS);
                    SerializedAsyncTaskProcessor.this.mCurrentTask = serializedAsyncTaskPoll;
                    if (serializedAsyncTaskPoll != null) {
                        notifyUI(0, serializedAsyncTaskPoll);
                        serializedAsyncTaskPoll.process();
                        notifyUI(1, serializedAsyncTaskPoll);
                    } else if (SerializedAsyncTaskProcessor.this.mKeepAliveTime > 0) {
                        SerializedAsyncTaskProcessor.this.stopTaskProcessor();
                    }
                } catch (InterruptedException e) {
                    MyLog.e(e);
                }
            }
        }
    }

    /* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/channel/commonutils/misc/SerializedAsyncTaskProcessor$SerializedAsyncTask.class */
    public static abstract class SerializedAsyncTask {
        public void postProcess() {
        }

        public void preProcess() {
        }

        public abstract void process();
    }

    public SerializedAsyncTaskProcessor() {
        this(false);
    }

    public SerializedAsyncTaskProcessor(boolean z) {
        this(z, 0);
    }

    public SerializedAsyncTaskProcessor(boolean z, int i) {
        this.mMainThreadHandler = null;
        this.threadQuit = false;
        this.mKeepAliveTime = 0;
        this.mMainThreadHandler = new Handler(Looper.getMainLooper()) { // from class: com.xiaomi.channel.commonutils.misc.SerializedAsyncTaskProcessor.1
            @Override // android.os.Handler
            public void handleMessage(Message message) {
                SerializedAsyncTask serializedAsyncTask = (SerializedAsyncTask) message.obj;
                if (message.what == 0) {
                    serializedAsyncTask.preProcess();
                } else if (message.what == 1) {
                    serializedAsyncTask.postProcess();
                }
                super.handleMessage(message);
            }
        };
        this.mIsDaemon = z;
        this.mKeepAliveTime = i;
    }

    /* JADX INFO: Access modifiers changed from: private */
    public void stopTaskProcessor() {
        synchronized (this) {
            this.mProcessThread = null;
            this.threadQuit = true;
        }
    }

    public void addNewTask(SerializedAsyncTask serializedAsyncTask) {
        synchronized (this) {
            if (this.mProcessThread == null) {
                ProcessPackageThread processPackageThread = new ProcessPackageThread();
                this.mProcessThread = processPackageThread;
                processPackageThread.setDaemon(this.mIsDaemon);
                this.threadQuit = false;
                this.mProcessThread.start();
            }
            this.mProcessThread.insertTask(serializedAsyncTask);
        }
    }

    public void addNewTaskWithDelayed(final SerializedAsyncTask serializedAsyncTask, long j) {
        this.mMainThreadHandler.postDelayed(new Runnable() { // from class: com.xiaomi.channel.commonutils.misc.SerializedAsyncTaskProcessor.2
            @Override // java.lang.Runnable
            public void run() {
                SerializedAsyncTaskProcessor.this.addNewTask(serializedAsyncTask);
            }
        }, j);
    }

    public void clearTask() {
        ProcessPackageThread processPackageThread = this.mProcessThread;
        if (processPackageThread != null) {
            processPackageThread.mTasks.clear();
        }
    }

    public void destroy() {
        this.threadQuit = true;
    }

    public SerializedAsyncTask getCurrentTask() {
        return this.mCurrentTask;
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/channel/commonutils/misc/SerializedAsyncTaskProcessor.kt
package com.xiaomi.channel.commonutils.misc

import android.os.Handler
import android.os.Looper
import android.os.Message
import com.xiaomi.channel.commonutils.logger.MyLog
import java.util.concurrent.LinkedBlockingQueue
import java.util.concurrent.TimeUnit

/*
 * Current override reference: com.xiaomi.xmsf 0.3.17-20260410000745 (versionCode 1003003000),
 * base.apk sha256 f3d72b6f5e1427ceecd3147a051d58e4dc95bb528397d486658e01cad9f7e590,
 * JADX path: com.xiaomi.xmsf/current/base/sources/com/xiaomi/channel/commonutils/misc/SerializedAsyncTaskProcessor.java
 */
class SerializedAsyncTaskProcessor @JvmOverloads constructor(
    private val mIsDaemon: Boolean = false,
    private var mKeepAliveTime: Int = 0,
) {
    @Volatile
    private var mCurrentTask: SerializedAsyncTask? = null
    private val mMainThreadHandler: Handler
    private var mProcessThread: ProcessPackageThread? = null
    @Volatile
    private var threadQuit = false

    private inner class ProcessPackageThread : Thread(THREAD_NAME) {
        val mTasks: LinkedBlockingQueue<SerializedAsyncTask> = LinkedBlockingQueue()

        private fun notifyUI(what: Int, task: SerializedAsyncTask) {
            try {
                mMainThreadHandler.sendMessage(mMainThreadHandler.obtainMessage(what, task))
            } catch (e: Exception) {
                MyLog.e(e)
            }
        }

        fun insertTask(task: SerializedAsyncTask) {
            try {
                mTasks.add(task)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }

        override fun run() {
            val keepAlive = if (mKeepAliveTime > 0) mKeepAliveTime.toLong() else Long.MAX_VALUE
            while (!threadQuit) {
                try {
                    val task = mTasks.poll(keepAlive, TimeUnit.SECONDS)
                    mCurrentTask = task
                    if (task != null) {
                        notifyUI(MSG_BEFORE_EXECUTE, task)
                        task.process()
                        notifyUI(MSG_AFTER_EXECUTE, task)
                    } else if (mKeepAliveTime > 0) {
                        stopTaskProcessor()
                    }
                } catch (e: InterruptedException) {
                    MyLog.e(e)
                }
            }
        }
    }

    abstract class SerializedAsyncTask {
        open fun postProcess() {
        }

        open fun preProcess() {
        }

        abstract fun process()
    }

    init {
        mMainThreadHandler = object : Handler(Looper.getMainLooper()) {
            override fun handleMessage(message: Message) {
                val task = message.obj as SerializedAsyncTask
                if (message.what == MSG_BEFORE_EXECUTE) {
                    task.preProcess()
                } else if (message.what == MSG_AFTER_EXECUTE) {
                    task.postProcess()
                }
                super.handleMessage(message)
            }
        }
    }

    fun stopTaskProcessor() {
        synchronized(this) {
            mProcessThread = null
            threadQuit = true
        }
    }

    fun addNewTask(serializedAsyncTask: SerializedAsyncTask) {
        synchronized(this) {
            if (mProcessThread == null) {
                val processThread = ProcessPackageThread()
                mProcessThread = processThread
                processThread.isDaemon = mIsDaemon
                threadQuit = false
                processThread.start()
            }
            mProcessThread?.insertTask(serializedAsyncTask)
        }
    }

    fun addNewTaskWithDelayed(serializedAsyncTask: SerializedAsyncTask, delay: Long) {
        mMainThreadHandler.postDelayed({ addNewTask(serializedAsyncTask) }, delay)
    }

    fun clearTask() {
        mProcessThread?.mTasks?.clear()
    }

    fun destroy() {
        threadQuit = true
    }

    fun getCurrentTask(): SerializedAsyncTask? = mCurrentTask

    companion object {
        private const val MSG_AFTER_EXECUTE = 1
        private const val MSG_BEFORE_EXECUTE = 0
        private const val THREAD_NAME = "PackageProcessor"
    }
}
